hca/util/__init__.py

```diff
@@ -302,6 +303,47 @@ def login(self, access_token=""):
                                         token_type="Bearer")
         print("Storing access credentials")
 
+    def _atomic_write_to_file(self, file, dir, content):
+        """
+        Atomically writes content a to file.
+
+        Creates a temp file and renames file (atomic operation) to desired filename.
+        To ensure atomicity, the src and dest dirs must be the same.
+
+        :param file: File name only (e.g. swagger.json)
+        :param dir: Shared directory of temp and dest file
+        :param content: Content to write to file
+        """
+        temp_filename = ""
+        dest_filename = os.path.join(dir, file)
+        writing = True
+        max_retries = 5
+        retry_count = 0
+        try:
+            with tempfile.NamedTemporaryFile('wb', dir=dir, delete=False) as temp:
+                temp_filename = temp.name
+                temp.write(content)
+            while writing and retry_count < max_retries:
```

Comment:
suggestion:

```
for _ in range(max_retries):
    try:
        os.rename(temp_filename, dest_filename)
        break
    except OSError as ex:
        if ex.errno == errno.EEXIST:
        # handle Windows: https://docs.python.org/2/library/os.html#os.rename
            os.remove(dest_filename)
            continue
        raise
else:
    raise Exception("Could not rename {} to {}".format(temp_filename, dest_filename))
```

you should be able to get rid of `writing` and `retry_count`

Reply:
Right. Python's a little rusty, thanks!